src/lib.rs

```diff
@@ -312,6 +321,18 @@ fn write_response<W>(response: Response, writer: &mut W) -> io::Result<()>
         Response::String(s) => {
             write!(writer, "${}#{:02x}", s, compute_checksum(s.as_bytes()))
         }
+        Response::Bytes(bytes) => {
+            // This would all be simpler if we wrapped |writer| with
+            // something to handle packets and checksumming.
```

Comment:
This sounds like a pretty useful thing. It could be a wrapper type that implements `Write`, or it could be as simple as `fn write_with_checksum<T: AsRef<&[u8]>(data: T)`.